apps/frontend/src/libs/modules/store/libs/middlewares/error-handling.middleware.ts

```diff
@@ -0,0 +1,17 @@
+import { isRejectedWithValue, type Middleware } from "@reduxjs/toolkit";
+import { toast } from "react-toastify";
+
+import { type AsyncThunkRejectValue } from "~/libs/types/types.js";
+
+const middleware: Middleware = () => {
+	return (next) => (action) => {
+		if (isRejectedWithValue(action)) {
+			const rejectedValue = action.payload as AsyncThunkRejectValue;
+			toast.error(rejectedValue.message);
```

Comment:
Let's use `isRejected` instead of `isRejectedWithValue`, then you should be able to access an error message from `action.error.message`. This way we will handle all the errors and we don't need to wrap each action with try/catch block

Reply:
Yeah. I thought about this solution, but it has it's limitations: `action.error` property has only `name`, `message` and `stack`, so we lost some context of an error (such as `status` for `HttpError`). So I decide to introduce `RejectValue` type, where we can specify, which fields we need for error to be displayed. However, I put there only `message` property right now... so ok, I will use your approach, (esspecially if take a count, that it reduces try/catch boiler-plate code)

Reply:
We will use custom names in case we need to handle errors differently